### DISCUSSION TITLE
KM and KM/H used instead of Miles and MPH

### DISCUSSION THREAD

**Discussion by pacemaker82** [alandtse/tesla#852] [General]
I am based in UK, we use miles, mph

My tesla is set to miles and mph. My HA is localised to UK too. 

What should I do to reflect the correct unit of measurement? Surely changing the unit on the sensor will take the raw km value (lets say 200km range) and change it to 200 miles, which would be incorrect.

**Comment by skipishere** [comment]
You can set the unit of measurement in the UI. Click on the sensor and then change the units. (It will correctly convert so you won't end up with 200 km to 200 miles)
![image](https://github.com/alandtse/tesla/assets/26724777/ec00680b-3092-4486-9c2b-0117d9c916a3)

**Reply by pacemaker82** [reply]
and that will convert the KM to Miles? Or take whatever value is in there and just change the unit after it?

**Reply by skipishere** [reply]
The backing information will be in km but all the sensor information (including what you see on the UI/triggers) will be the converted to miles (so 200 km would be 124.2~ish miles). Try it and you'll see

**Reply by pacemaker82** [reply]
thanks! it worked as you described - appreciate it :) 

Is there a way to prepend all the sensors/entities with "tesla" ... the naming structure is very generic and wonder what happens if it clashes with other entities I might have... example: "sensor.battery" could be anything, would prefer "sensor.tesla.battery" or something that makes it unique to the car... "switch.charger" "update.softwareupdate" ... these arent very unique and hard to search for when you have other battery stuff going on on your HA

**Reply by skipishere** [reply]
By default your sensors/switches/etc should all be using <sensor/switch/domain>.<name of car in app>_<thing> so that they don't potentially clash (Home Assistant will smartly add a 2,3,4,etc if there is a name collision for any reason). I did go through and rename mine as I like having my car's name with my registration plate in the app (cause I can't remember it when I need to pay for parking, lol) but don't want it in my Home Assistant UI.

**Reply by pacemaker82** [reply]
got it thank you. hopefully last Q - is there a way to see the energy charged today? The "sensor.energy_added" goes up and down if you plug in / unplug (basically each session seems to reset it to 0). I am looking for a total charged kWh for each day
